### DISCUSSION TITLE
wire:click isn't working sometime ?

### DISCUSSION THREAD

**Discussion by Yondalne** [livewire/livewire#4730] [1. Help] [answered]
Hello, i'm starting work with livewire recently and want to make a dependant dropdown but the wire:click isn't working on my component and i do not have any issue so ....

That's the code of the component
`class Category extends Component {
    public $allCategories = [];
    public $subCats = [];
    
    public function setSubcat() {
        $this->subCats = [];
    
        $subCat = SubCategory::where("category_id", 1)->get();
        foreach ($subCat as $sub){
            array_push($this->subCats, $sub);
        }
        $this->render();
    }
    
    public function mount() {
        $this->allCategories = ModelsCategory::all();
        // $this->subCats = SubCategory::all();
    }
    
    public function render()  {
        return view('livewire.category');
    }
}`

And here the blade file
![image](https://user-images.githubusercontent.com/83779824/155691639-b041eec6-6901-491d-a4a5-4d4eec73edb6.png)

If you know also an other method to do that, let me know. Thanks

**Comment by silnarm** [comment, marked as answer]
This is a strange use of wire:click, and while I don't know for sure I suspect having that on an option is causing the issue (probably shouldn't of course, but as I said this is strange usage and it is unlikely that would have been tested).

You would normally model a select, with a 'selected id' prop on the component.

If you wish to try this to see if your problem goes away, remove/comment you setSubcat() method and add this in the component:
```php
    public $selectedCategoryId = 0;

    public function updatedSelectedCategoryId($value) {
        $this->subCats = SubCategory::where("category_id", $this->selectedCategoryId)->get()->toArray();
    }
```
and in your blade `wire:model` the select and remove the `wire:click` attribs of the options:
```blade
            <select wire:model="selectedCategoryId">
                <option value="0" disabled>Select...</option>
                @foreach ($allCategories as $category)
                    <option value="{{ $category->id }}">{{ $category->title }}</option>
                @endforeach
            </select>
```

**Reply by Yondalne** [reply]
Thank you that's working.
But now i'm a kind curious to know why with wire:click that's not working 🤔

**Reply by joshhanley** [reply]
@Yondalne it looks like you are using select2, if it were a standard select I think `wire:click` would work, but it's not recommended. The best way is as suggested by @silnarm.

**Reply by Yondalne** [reply]
Oh okay
i will try to see if that's the problem, thanks for response

**Comment by Yondalne** [comment]
Sorry about my code ... as i said i'm a beginner with.
Thank you for help i'm going to try that
